lib/OpenQA/WebSockets/Controller/Worker.pm

```diff
@@ -234,4 +198,47 @@ sub _message ($self, $json) {
     }
 }
 
+sub _should_assign_jobs ($self, $worker, $worker_id, $current_job_id, $job_id, $job_token, $worker_previously_idle) {
+    return undef unless $worker;
```

Comment:
I think I can remove this due to https://github.com/os-autoinst/openQA/pull/6228/files#diff-51176bd675ee4037fef74049369705b9259134ad87070a88e59501936761fee1R163. What do you think?

Reply:
or even refactor further this part and combine it with the try block of that reference

Reply:
Why do you think it can be removed due to the line you linked to?
That's a 
```
if ($worker) {
...
}
```
block above

Reply:
the idea was that the there is an earlier check of the $worker. The simplest move will be to move the code inside this `if` block. or as you did I think in your PR return earlier.

Reply:
And combining the two try blocks to one would also not work, as they are catching two different problems.

Reply:
can this resolved as the https://github.com/os-autoinst/openQA/pull/6238 merged and it might be different situation now

Reply:
Well, sure. You opened this, but I' can resolve it as well